### PR TITLE
Update babel to v7.20.0

### PR DESCRIPTION
**Pull request by fisker**
## Description

Babel going to make some change to the location info https://github.com/babel/babel/pull/15032. I downloaded the artifact to see what will break.

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Note to myself, `babel-ts` range maybe incorrect.

File: src/language-js/loc.js [hunk 1]
```diff
@@ -4,20 +4,16 @@ import isNonEmptyArray from "../utils/is-non-empty-array.js";
...
-  if (!ignoreDecorators) {
-    const decorators =
-      (node.declaration && node.declaration.decorators) || node.decorators;
-
-    if (isNonEmptyArray(decorators)) {
-      return locStart(decorators[0]);
-    }
+  const decorators = node.declaration?.decorators ?? node.decorators;
+  if (isNonEmptyArray(decorators)) {
+    return Math.min(locStart(decorators[0]), start);
```
Comment (fisker) [inline src/language-js/loc.js:13]: It seems the old one is not right, if decorators after `export`, it will still use range of `decorators[0]`

**Comment by fisker** [comment]
This PR only fixes `hasDecoratorsBeforeExport` check, other changes like

- Parse import reflection
- Add support for expr satisfies Type expressions

Will be done in separate prs.

https://github.com/babel/babel/releases/tag/v7.20.0

**Review by sosukesuzuki — APPROVED** [review]

File: src/language-js/loc.js (hunk 1 above)
Comment (nicolo-ribaudo) [reply src/language-js/loc.js:13]: In which case is `locStart(decorators[0])` less then `start`?

Comment (fisker) [reply src/language-js/loc.js:13]: > In which case is locStart(decorators[0]) less then start?

Other parsers, `flow`/`typescript` I think they followed old behavior of babel.

**Pull request merged**